the-collab-lab/tcl-16-smart-shopping-list
# Resolves 10. As an item, I want my estimated next purchase date to be computed at the time my purchase is recorded in the database so the app can learn how often I buy different items.

## Description

This code mainly deals with how we are writing to the Firebase database. When a user adds an item to their shopping list we are also recording the ```numberOfPurchases``` to keep track of how many times a user has purchased the item. Now when a user checks an item off their list these three things are updated:
*  ```daysToPurchase```, which based on ```estimates.js``` is the ```lastEstimate``` and is the return of ```calculateEstimate```
* ```lastPurchasedDate```, which updates to the time the user checks the item off
* ```numberOfPurchases``` , which increases by 1 each time the user checks the item off

## Related Issue

Closes #10 

## Acceptance Criteria

* When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

<img width="538" alt="Screen Shot 2021-02-04 at 11 08 00 AM" src="https://user-images.githubusercontent.com/62153993/106922303-ea9d1980-66da-11eb-8924-20fe61f1b3ad.png">

## Testing Steps / QA Criteria
* switch to our branch ```WL-JS-estimate-next-purchase-date```
* add item to list
* go to Firebase and change the ```lastPurchasedDate``` of the item to before the date you are testing
* check off the same item
* in Firebase you should see the item's ```daysToPurchase```, ```lastPurchasedDate```, and ```numberOfPurchases``` updated
